controllers/auth.js

```diff
@@ -9,6 +9,81 @@ const {
   USER_DOES_NOT_EXIST_ERROR,
 } = require("../constants/errorMessages");
 
+const googleAuthLogin = (req, res, next) => {
+  const { redirectURL, dev } = req.query;
+  if (dev === "true") {
+    return passport.authenticate("google", {
+      scope: ["email"],
+      state: redirectURL,
+    })(req, res, next);
+  } else {
```

Comment:
if we remove the else block without handling the case where the dev query parameter is missing, the request will be left hanging as this function is directly used here
`router.get("/google/login", auth.googleAuthLogin);`

Reply:
How so? 
won't it be the same, if you add else or not?

Reply:
(Although now i have added devFlag middleware so this is not required but still wanted feedback)

@prakashchoudhary07  I’m not entirely clear on how it would behave same. My understanding is that if we remove the `else` block and don't explicitly handle the case where the `dev` query parameter is missing, the request could continue executing without sending a response, potentially leaving the user in a waiting state. This could lead to an unhandled request where the user doesn’t get any feedback or proper handling for the missing parameter.